srml/metadata/src/lib.rs

```diff
@@ -333,7 +336,7 @@ impl Decode for RuntimeMetadataDeprecated {
 /// The metadata of a runtime version 2.
 #[derive(Eq, Encode, PartialEq)]
 #[cfg_attr(feature = "std", derive(Decode, Debug, Serialize))]
-pub struct RuntimeMetadataV2 {
+pub struct RuntimeMetadataV3 {
```

Comment:
Should we rename to something neutral (update of version is already done in the enum). I am also starting to realize choice of enum to encode version can be cumbersome (lot of code change), I should have use struct{version, inner}.

Reply:
ok, but I don't know which name, as RuntimeMetadata is used, maybe RuntimeMetadataImpl but hmm I don't know

Reply:
I am bad at naming things 'RuntimeMetadataCurrent' as opposed to 'RuntimeMetadataDeprecated'?
(no need to change the enum in this pr it was more a general remark (it may require a handwritten codec to be aligned on current behavior (not sure)))